lib/src/connector/course_connector.dart

```diff
@@ -101,38 +101,57 @@ class CourseConnector {
       //取得學期資料
       nodes = courseNodes[0].getElementsByTagName("td");
       SemesterJson semester = SemesterJson();
-      semester.year = nodes[1].text;
-      semester.semester = nodes[2].text;
+
+      // the title String of the first course table was stored seperately in its <td> element,
+      // but it currently stores all the information in a row,
+      // (ex: 學號：110310144　　姓名：xxx　　班級：電機三甲　　　 112 學年度 第 1 學期　上課時間表)
+      // so the RegExp is used to filter out only the number parts
+      final titleString = nodes[0].text;
+      final RegExp studentSemesterDetailFilter = RegExp(r'\b\d+\b');
+      final Iterable<RegExpMatch> studentSemesterDetailMatches = studentSemesterDetailFilter.allMatches(titleString);
+      // "studentSemesterDetails" should consist of three numerical values
+      // ex: [110310144, 112, 1]
+      final List<String> studentSemesterDetails = studentSemesterDetailMatches.map((match) => match.group(0)).toList();
+      if (studentSemesterDetails.isEmpty) {
+        throw RangeError("[TAT] course_connector.dart: studentSemesterDetails list is empty");
+      }
+      if (studentSemesterDetails.length < 3) {
+        throw RangeError("[TAT] course_connector.dart: studentSemesterDetails list has range less than 3");
+      }
+      semester.year = studentSemesterDetails[1];
+      semester.semester = studentSemesterDetails[2];
 
       courseExtraInfo.courseSemester = semester;
 
       CourseExtraJson courseExtra = CourseExtraJson();
 
-      courseExtra.name = nodes[3].getElementsByTagName("a")[0].text;
-      if (nodes[3].getElementsByTagName("a")[0].attributes.containsKey("href")) {
-        courseExtra.href = _courseCNHost + nodes[3].getElementsByTagName("a")[0].attributes["href"];
+      nodes = courseNodes[1].getElementsByTagName("tr");
+      final List<String> courseIds = nodes.skip(2).map((node) => node.getElementsByTagName("td")[0].text).toList();
+      final courseIdPosition = courseIds.indexWhere((element) => element.contains(courseId));
+      if (courseIdPosition == -1) {
+        throw StateError('[TAT] course_connector.dart: CourseId not found: $courseId');
+      } else {
+        node = nodes[courseIdPosition + 2];
       }
-      courseExtra.category = nodes[7].text; // 取得類別
-      courseExtra.openClass = nodes[9].text;
-      courseExtra.selectNumber = nodes[11].text;
-      courseExtra.withdrawNumber = nodes[12].text;
-      courseExtra.id = courseId;
+      classExtraInfoNodes = node.getElementsByTagName("td");
+      courseExtra.id = strQ2B(classExtraInfoNodes[0].text).replaceAll(RegExp(r"[\n| ]"), "");
```

Comment:
```suggestion
      courseExtra.id = strQ2B(classExtraInfoNodes[0].text).replaceAll(RegExp(r"\s"), "");
```

Using `\s` here can be easier and more reliable.